ocs_ci/deployment/aws.py

```diff
@@ -0,0 +1,155 @@
+"""
+This module contains platform specific methods and classes for deployment
+on AWS platform
+"""
+import os
+import logging
+import pytest
+import json
+import traceback
+
+from .deployment import Deployment
+from ocs_ci.utility.utils import run_cmd
+from ocs_ci.ocs.openshift_ops import OCP
+from ocs_ci.framework import config
+from ocs_ci.ocs.parallel import parallel
+from ocs_ci.utility.aws import AWS as AWSUtil
+from ocs_ci.utility import utils
+
+
+logger = logging.getLogger(name=__file__)
+
+
+# As of now only ipi
+# TODO: Introduce UPI once we have proper doc
+__all__ = ['AWSIPI']
+
+
+class AWSBase(Deployment):
+    def __init__(self):
+        """
+        This would be base for both ipi and upi deployment
+        """
+        super(AWSBase, self).__init__()
+        self.region = config.ENV_DATA['region']
+        self.aws = AWSUtil(self.region)
+
+    def create_ebs_volumes(self, worker_pattern, size=100):
+        """
+        Add new ebs volumes to the workers
+
+        Args:
+            worker_pattern (str):  Worker name pattern e.g.:
+                cluster-55jx2-worker*
+            size (int): Size in GB (default: 100)
+        """
+        worker_instances = self.aws.get_instances_by_name_pattern(
+            worker_pattern
+        )
+        with parallel() as p:
+            for worker in worker_instances:
+                logger.info(
+                    f"Creating and attaching {size} GB "
+                    f"volume to {worker['name']}"
+                )
+                p.spawn(
+                    self.aws.create_volume_and_attach,
+                    availability_zone=worker['avz'],
+                    instance_id=worker['id'],
+                    name=f"{worker['name']}_extra_volume",
+                    size=size,
+                )
+
+    def add_volume(self, size=100):
+        """
+        Add a new volume to all the workers
+
+        Args:
+            size (int): Size of volume in GB (default: 100)
+        """
+        with open(os.path.join(self.cluster_path, "terraform.tfvars")) as f:
+            tfvars = json.load(f)
+
+        cluster_id = tfvars['cluster_id']
+        worker_pattern = f'{cluster_id}-worker*'
+        logger.info(f'Worker pattern: {worker_pattern}')
+        self.create_ebs_volumes(worker_pattern, size)
+
+    def add_node(self):
+        # TODO: Implement later
```

Comment:
I just wanted it to be there so that its bit clear on what are the possible methods that could be applicable to this class, ofcourse it can be referred from base class as well.
I am ok with both